gnd/src/main/java/com/google/android/gnd/ui/map/gms/GoogleMapsFragment.java

```diff
@@ -480,6 +480,23 @@ public void setMapFeatures(ImmutableSet<MapFeature> features) {
     }
   }
 
+  @Override
+  public void refreshMarkers() {
+    Set<MapPin> mapPinsToAdd = new HashSet<>();
+    for(Marker marker: markers) {
+      MapPin mapPin = (MapPin) marker.getTag();
+
+      if (mapPin != null) {
+        mapPinsToAdd.add(mapPin);
+        removeMarker(marker);
+      }
+    }
+
+    for(MapPin mapPin: mapPinsToAdd) {
+      addMapPin(mapPin);
+    }
```

Comment:
Should we just update the icon instead recreating the markers?

Reply:
Markers are immutable, so I believe that's not possible:
https://developers.google.com/android/reference/com/google/android/gms/maps/model/Marker

Reply:
Verified that marker icon can be mutated if you have reference to the marker.

In this test, I update the color of already rendered marker whenever it is clicked. 
```
  private boolean onMarkerClick(Marker marker) {
    marker.setIcon(
        markerIconFactory.getMarkerIcon(getResources().getColor(android.R.color.holo_blue_bright)));
 }
```

Reply:
Great idea, done!